src/js/abstract/base-component-global.js

```diff
@@ -19,19 +20,33 @@ export default class BaseComponentGlobal extends BaseComponent {
    * append the custom element into the dom
    *
    * @param  {type} styles description
-   * @param  {type} nodeName description
+   * @param  {type} [nodeName=UUID] description
    * @return {type}        description
    */
   static appendGlobalStyles(styles, nodeName = BaseComponent.uuidv4()) {
-    if (styles) {
-      if (!memory[styles]) {
-        const styleNode = document.createElement('style');
-        const styleText = document.createTextNode(styles);
-        styleNode.appendChild(styleText);
-        styleNode.setAttribute('data-c-name', nodeName.toLowerCase());
-        document.querySelector('head').insertAdjacentElement('afterbegin', styleNode);
-        memory[styles] = true;
+    if (styles && !memory[nodeName]) {
+      let target = document.head;
+      const styleNode = document.createElement('style');
+      const styleText = document.createTextNode(styles);
+
+      memory[nodeName] = true;
+
+      styleNode.appendChild(styleText);
+      styleNode.setAttribute('data-c-name', nodeName.toLowerCase());
+
+      // It's super important that each components CSS is evaluated after our reboot.css
+      // Especially regarding integration this should prevent breaking our cascade.
+      if (!styleTags) {
+        styleTags = document.getElementsByTagName('style');
       }
+
+      const { length } = styleTags;
+
+      if (length) {
+        target = styleTags[length - 1].parentNode;
```

Comment:
@LucaMele 
I see only one possible issue remaining.
What if the last stylesheet's tag `<style>` is after any custom element `<axa-foo>`.
Because our custom elements using global styles need their CSS before their usage.
So we may need special handling if such a case exists - keyword AEM Clientlibs and Design Importer?